src/main/scala/org/kframework/kore/interfaces/Interface.scala

```diff
@@ -0,0 +1,303 @@
+  }
+
+  object SymbolDeclaration {
+    def unapply(arg: SymbolDeclaration): Option[(Sort, Symbol, Seq[Sort], Attributes)] = Some(arg.sort, arg.symbol, arg.args, arg.att)
+  }
+
+
+  trait Rule extends Sentence {
+    val p: Pattern
+  }
+
+  object Rule {
+    def unapply(arg: Rule): Option[(Pattern, Attributes)] = Some(arg.p, arg.att)
+  }
+
+  trait Axiom extends Sentence {
+    val p: Pattern
+  }
+
+  object Axiom {
+    def unapply(arg: Axiom): Option[(Pattern, Attributes)] = Some(arg.p, arg.att)
+  }
+
+  trait Attributes {
+    def att: Seq[Pattern]
+  }
+
+  object Attributes {
+    def unapply(arg: Attributes): Option[Seq[Pattern]] = Some(arg.att)
+  }
+
+
+  sealed trait Pattern
+
+  trait Variable extends Pattern {
+    val name: Name
+    val sort: Sort
+  }
+
+  object Variable {
+    def unapply(arg: Variable): Option[(Name, Sort)] = Some(arg.name, arg.sort)
+  }
+
+  trait Application extends Pattern {
+    val symbol: Symbol
+    val args: Seq[Pattern]
+  }
+
+  object Application {
+    def unapply(arg: Application): Option[(Symbol, Seq[Pattern])] = Some(arg.symbol, arg.args)
+  }
+
+  trait DomainValue extends Pattern {
+    val symbol: Symbol
+    val value: Value
+  }
+
+  object DomainValue {
+    def unapply(arg: DomainValue): Option[(Symbol, Value)] = Some(arg.symbol, arg.value)
+  }
+
+  trait Top extends Pattern
+
+  object Top {
+    def unapply(arg: Top): Boolean = true
+  }
+
+  trait Bottom extends Pattern
+
+  object Bottom {
+    def unapply(arg: Bottom): Boolean = true
+  }
+
+  trait And extends Pattern {
+    val p: Pattern
+    val q: Pattern
+  }
+
+  object And {
+    def unapply(arg: And): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
+  }
+
+  trait Or extends Pattern {
+    val p: Pattern
+    val q: Pattern
+  }
+
+  object Or {
+    def unapply(arg: Or): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
+  }
+
+  trait Not extends Pattern {
+    def p: Pattern
+  }
+
+  object Not {
+    def unapply(arg: Not): Option[Pattern] = Some(arg.p)
+  }
+
+  trait Implies extends Pattern {
+    val p: Pattern
+    val q: Pattern
+  }
+
+  object Implies {
+    def unapply(arg: Implies): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
+  }
+
+  trait Exists extends Pattern {
+    val v: Variable
+    val p: Pattern
+  }
+
+  object Exists {
+    def unapply(arg: Exists): Option[(Variable, Pattern)] = Some(arg.v, arg.p)
+  }
+
+  trait ForAll extends Pattern {
+    val v: Variable
+    val p: Pattern
+  }
+
+  object ForAll {
+    def unapply(arg: ForAll): Option[(Variable, Pattern)] = Some(arg.v, arg.p)
+  }
+
+  trait Next extends Pattern {
+    def p: Pattern
+  }
+
+  object Next {
+    def unapply(arg: Next): Option[Pattern] = Some(arg.p)
+  }
+
+
+  trait Rewrite extends Pattern {
+    val p: Pattern
+    val q: Pattern
+  }
+
+  object Rewrite {
+    def unapply(arg: Rewrite): Option[(Pattern, Pattern)] = Some(arg.p, arg.q)
+  }
+
+  trait Equals extends Pattern {
+    val p: Pattern
+    val q: Pattern
```

Comment:
Ok. I've changed it to _1, and _2 everywhere except the quantifiers, where v, and p remain, representing Variable and Pattern respectively. Given that this doesn't have any relation to the Kore language itself, I'm ok with this decision, if it helps make things easier in Kale.